openehr-rm/src/main/java/com/nedap/archie/rminfo/ArchieRMInfoLookup.java

```diff
@@ -169,6 +174,25 @@ protected void addTypes(Class baseClass) {
         addClass(Section.class);
         addClass(Activity.class);
         addClass(TerminologyId.class);
+        addClass(Link.class);
+        addClass(Archetyped.class);
+        addClass(ArchetypeHRID.class);
+        addClass(AuthoredResource.class);
+        addClass(TranslationDetails.class);
+    }
+
+    @Override
+    protected boolean isNullable(Class clazz, Method getMethod, Field field) {
+        if(field != null) {
+            if (Party.class.isAssignableFrom(clazz) && field.getName().equalsIgnoreCase("uid")) {
+                return false;
+            }
+        } else if (getMethod != null) {
+            if (Party.class.isAssignableFrom(clazz) && getMethod.getName().equalsIgnoreCase("getUid")) {
+                return false;
+            }
+        }
```

Comment:
Why are these special cases needed? Maybe add a comment.

Reply:
good call. They are needed because Party overrides the nulltable from Actor, and that is not possible to model in java without jumping through major hoops :)

Reply:
I'll add comment